codersdk/deploymentconfig.go

```diff
@@ -108,13 +108,14 @@ type TLSConfig struct {
 }
 
 type GitAuthConfig struct {
-	ID           string `json:"id"`
-	Type         string `json:"type"`
-	ClientID     string `json:"client_id"`
-	ClientSecret string `json:"-" yaml:"client_secret"`
-	AuthURL      string `json:"auth_url"`
-	TokenURL     string `json:"token_url"`
-	Regex        string `json:"regex"`
+	ID           string   `json:"id"`
+	Type         string   `json:"type"`
+	ClientID     string   `json:"client_id"`
+	ClientSecret string   `json:"-" yaml:"client_secret"`
+	AuthURL      string   `json:"auth_url"`
+	TokenURL     string   `json:"token_url"`
+	Regex        string   `json:"regex"`
+	Scopes       []string `json:"scopes"`
```

Comment:
Should we consider `omitempty` here? I noticed typegen needs to be run for this PR, which adds the TypeScript entry for `scopes`, however, if the API doesn't ensure that `Scopes: nil` becomes, `Scopes: []string{}`, then we're lying to TypeScript (at which point using `omitempty` is appropriate).

Reply:
Ahh, good point!